go-controller/pkg/ovn/pods_test.go

```diff
@@ -400,27 +400,156 @@ var _ = ginkgo.Describe("OVN Pod Operations", func() {
-				gomega.Expect(success).To(gomega.BeFalse())
+				err := fakeOvn.controller.ensurePod(nil, podObj, true) // this fails since pod doesn't exist to set annotations
+				gomega.Expect(err).To(gomega.HaveOccurred())
 
 				gomega.Expect(len(fakeOvn.controller.retryPods)).To(gomega.Equal(0))
 				fakeOvn.controller.addRetryPods([]v1.Pod{*podObj})
+				key := getPodNamespacedName(podObj)
 				gomega.Expect(len(fakeOvn.controller.retryPods)).To(gomega.Equal(1))
-				gomega.Expect(fakeOvn.controller.retryPods["123"]).ToNot(gomega.BeNil())
-				gomega.Expect(fakeOvn.controller.retryPods["123"].ignore).To(gomega.BeFalse())
-				gomega.Expect(fakeOvn.controller.retryPods["123"].pod.UID).To(gomega.Equal(podObj.UID))
+				gomega.Expect(fakeOvn.controller.retryPods[key]).ToNot(gomega.BeNil())
+				gomega.Expect(fakeOvn.controller.retryPods[key].ignore).To(gomega.BeFalse())
+				gomega.Expect(fakeOvn.controller.retryPods[key].pod.UID).To(gomega.Equal(podObj.UID))
 
-				fakeOvn.controller.checkAndSkipRetryPod(podObj.UID)
-				gomega.Expect(fakeOvn.controller.retryPods["123"].ignore).To(gomega.BeTrue())
+				fakeOvn.controller.checkAndSkipRetryPod(podObj)
+				gomega.Expect(fakeOvn.controller.retryPods[key].ignore).To(gomega.BeTrue())
 
 				fakeOvn.controller.unSkipRetryPod(podObj)
-				gomega.Expect(fakeOvn.controller.retryPods["123"].ignore).To(gomega.BeFalse())
+				gomega.Expect(fakeOvn.controller.retryPods[key].ignore).To(gomega.BeFalse())
+
+				fakeOvn.controller.checkAndDeleteRetryPod(podObj)
+				gomega.Expect(fakeOvn.controller.retryPods[key]).To(gomega.BeNil())
+				return nil
+			}
+
+			err := app.Run([]string{app.Name})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		})
+
+		ginkgo.It("correctly retries a failure while adding a pod", func() {
+			app.Action = func(ctx *cli.Context) error {
+				namespace1 := *newNamespace("namespace1")
+				podTest := newTPod(
+					"node1",
+					"10.128.1.0/24",
+					"10.128.1.2",
+					"10.128.1.1",
+					"myPod",
+					"10.128.1.3",
+					"0a:58:0a:80:01:03",
+					namespace1.Name,
+				)
+				pod := newPod(podTest.namespace, podTest.podName, podTest.nodeName, podTest.podIP)
+
+				fakeOvn.startWithDBSetup(initialDB,
+					&v1.NamespaceList{
+						Items: []v1.Namespace{
+							namespace1,
+						},
+					},
+					&v1.PodList{
+						Items: []v1.Pod{*pod},
+					},
+				)
+
+				podTest.populateLogicalSwitchCache(fakeOvn, getLogicalSwitchUUID(fakeOvn.controller.nbClient, "node1"))
+				fakeOvn.controller.WatchNamespaces()
+				fakeOvn.asf.ExpectAddressSetWithIPs(podTest.namespace, []string{podTest.podIP})
+				gomega.Eventually(fakeOvn.controller.nbClient).Should(
+					libovsdbtest.HaveData(getExpectedDataPodsAndSwitches([]testPod{}, []string{"node1"})...))
+
+				// inject transient problem, nbdb is down
+				fakeOvn.controller.nbClient.Close()
+				gomega.Eventually(fakeOvn.controller.nbClient.Connected()).Should(gomega.BeFalse())
```

Comment:
This needs to be fix and sorry this was my fault. I improperly did this in the first round of network policy retry. I fixed these to take a function rather than a call in my last NP patch:
https://github.com/openshift/ovn-kubernetes/commit/7f32984be1439aedba682ea5bf5db963b4c98684#diff-2a5ea421d13f2f639ad217cada1f5c741e9ba6d68abcf431a700c0f68b71e04dR2176

Reply:
done.